SCClassLibrary/Common/Core/Clock.sc

```diff
@@ -327,6 +327,48 @@ elapsed time is whatever the system clock says it is right now. elapsed time is
 		^this.beats - this.bars2beats(this.bar)
 	}
 
+	sync { arg tempo, secs = 4, resolution = 1;
+		var next, time, durCur, durNew, durDif, durAvg, stepsPerBeat,
+			delta, factor, steps, stepSize, sum, lastDur, index = 0;
+
+		secs = secs.max(0.03);				// saftey and lower jitter limit
+		next = this.timeToNextBeat(1);
+		time = secs - (this.tempo.reciprocal * next);
+		if(time < next) {						// jump directly
+			this.tempo_(next / secs);		// set a high tempo
+			this.sched(next, { this.tempo_(tempo); nil });
+		} {							// else interpolate
+			this.sched(next, {		// start at next beat
+				durCur = this.tempo.reciprocal;
+				durNew = tempo.reciprocal;
+				durDif = durNew - durCur;
+				durAvg = durCur + durNew / 2;		// average duration for number of steps
+				stepsPerBeat = resolution.max(0.001).reciprocal.round;
+				steps = (time / durAvg).round * stepsPerBeat;
+				delta = stepsPerBeat.reciprocal;		// quantized resolution
+				stepSize = durDif / steps;
+				lastDur = (steps - 1 * stepSize) + durCur;
+				sum = durCur + lastDur / 2 * steps / stepsPerBeat; // sum of an arithmetic progression
+				factor = time / sum;
+				this.sched(0, {
+					var tmp;
+					if(index < steps) {
+						durCur = durCur + stepSize;
+						tmp = (durCur * factor).reciprocal;
+						this.tempo_(tmp);
```

Comment:
no this isn't possible because the line above changes `durCur` in each iteration.